intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
+\item  Takes care of \underbar{dependencies}: Another issue arises when one considers library versions and the dependencies they create. Some software requires an older version of a particular library to run correctly (or at all). Hence a variety of version numbers is available for important libraries.
+\end{enumerate}
+
+This is all managed with the ``\emph{module}'' command, which is explained in the next sections.
+
+\subsection{Available modules}
+
+A large number of software packages are installed on the \hpc clusters. A
+list of all currently available software can be obtained by typing:
+
+\begin{prompt}
+%\shellcmd{module av}%
+\end{prompt}
+
+or by typing it in full:
+
+\begin{prompt}
+%\shellcmd{module available}%
+\end{prompt}
+
+This will give some output such as:
+
+\inputsite{available-modules}
+
+This gives a full list of software packages that can be loaded. Note that
+modules starting with a capital letter are listed first.
+
+\subsection{Organisation of modules in toolchains}
+
+The amount of modules on the VSC systems can be overwhelming, and it is not
+always immediately clear which modules can be loaded safely together if you
+need to combine multiple programs in a single job to get your work done.
+
+Therefore the VSC has defined so-called \strong{toolchains} on the newer VSC-clusters.
+A toolchain contains a C/C++ and Fortran compiler, a MPI library and some basic
+math libraries for (dense matrix) linear algebra and FFT. Two toolchains are
+defined on most VSC systems. One, the \lstinline|intel| toolchain, consists of the Intel
+compilers, MPI library and math libraries. The other one, the \lstinline|foss| toolchain,
+consists of Open Source components: the GNU compilers, OpenMPI, OpenBLAS and
+the standard LAPACK and ScaLAPACK libraries for the linear algebra operations
+and the FFTW library for FFT. The toolchains are refreshed twice a year,
+which is reflected in their name.
+
+E.g., \lstinline|foss-2014a| is the first version of the \lstinline|foss| toolchain in 2014.
+
+The toolchains are then used to compile a lot of the software installed on
+the VSC clusters. You can recognise those packages easily as they all contain the
+name of the toolchain after the version number in their name. Packages
+compiled with the same toolchain and toolchain version will typically work
+together well without conflicts.
+
+For some clusters, additional toolchains are defined, e.g., to take advantage
+of specific properties of that cluster such as GPU accelerators or special
+interconnect features that require a vendor-specific MPI-implementation.
+
+\subsection{Activating and de-activating modules}
+\label{subsec:activating-and-deactivating-modules}
+
+A module is loaded using the following command:
+
+
+\begin{prompt}
+%\shellcmd{module load \emph{examplemodule}}%
+\end{prompt}
+
+This will load the most recent version of \emph{examplemodule}.
```

Comment:
Clarify that this is bad practice usually: you should  almost always load a *specific* version of a module, rather than just loading the default version (which usually corresponds to the latest version). The default version can change over time, and so you'll suddenly be loading a different software version, which can break your job script or result in different results...

Reply:
Ah, that's actually done below... Should maybe stand out more?